gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/dichotomy/DichotomyParallelization.java

```diff
@@ -49,15 +56,29 @@ public SweResponse launchDichotomy(SweData sweData) {
         // build swe response from every response
         String ttcDocUrl = outputService.buildAndExportTtcDocument(sweData, executionResult);
         String voltageEsFrZipUrl = outputService.buildAndExportVoltageDoc(DichotomyDirection.ES_FR, sweData, executionResult);
-        return  new SweResponse(sweData.getId(), ttcDocUrl, voltageEsFrZipUrl);
+        SweDichotomyResult esFrResult = getByDirection(executionResult, DichotomyDirection.ES_FR);
+        return  new SweResponse(sweData.getId(), ttcDocUrl, voltageEsFrZipUrl, esFrResult.getHighestValidStepUrl(), esFrResult.getLowestInvalidStepUrl());
     }
 
     SweDichotomyResult runDichotomyForOneDirection(SweData sweData, DichotomyDirection direction) {
         DichotomyResult<RaoResponse> dichotomyResult = dichotomyRunner.run(sweData, direction);
         dichotomyLogging.logEndOneDichotomy(direction);
         // Generate files specific for one direction (cne, cgm, voltage) and add them to the returned object (to create)
+        String highestValidStepUrl = null;
```

Comment:
I wish we would not overcrowd this method with low level stuff. Please extract the initialisation and the test.

Reply:
working on it ...